### PR TITLE
Set log level using env variable

### PR DESCRIPTION
**Pull request by kale-amruta**
Signed-off-by: Amruta Kale <amrutakale24.1991@gmail.com>

## Change Overview
Kanister by default logs at info level. This PR adds a way to change the log level setting

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by kale-amruta** [comment]
@PrasadG193 can you review it?

File: pkg/log/log.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ const (
 	InfoLevel Level = Level(logrus.InfoLevel)
 	// ErrorLevel log level.
 	ErrorLevel Level = Level(logrus.ErrorLevel)
+
+	// LevelVarName is the environment variable that controls
+	// init log levels
+	LevelVarName = "LOG_LEVEL"
```
Comment (PrasadG193) [inline pkg/log/log.go:31]:
```suggestion
	LevelEnvName = "LOG_LEVEL"
```

File: pkg/log/log.go [hunk 2]
```diff
@@ -121,6 +125,17 @@ func SetFormatter(format OutputFormat) {
 func init() {
 	SetFormatter(TextFormat)
 	initEnvVarFields()
+	level, err := logrus.ParseLevel(os.Getenv(LevelVarName))
+
+	if err != nil {
+		level = logrus.InfoLevel
+	}
```
Comment (PrasadG193) [inline pkg/log/log.go:132]: Should we move this under `SetLevel()`?

File: pkg/log/log_test.go [hunk 3]
```diff
@@ -105,3 +105,26 @@ func testLogMessage(c *C, msg string, print func(string, ...field.M), fields ...
...
+	var entry map[string]interface{}
+	//Check if debug level log is printed when log level is info
+	Debug().WithContext(ctx).Print("Testing debug level")
+	err := json.Unmarshal(output.Bytes(), &entry)
+
+	c.Assert(err, NotNil)
+	c.Assert(output.String(), HasLen, 0)
+
+	//Check if debug level log is printed when log level is debug
+	log.SetLevel(logrus.DebugLevel)
```
Comment (PrasadG193) [inline pkg/log/log_test.go:124]: The more accurate validation would be to verify if the log_level is getting set if `LOG_LEVEL` env var is set

File: pkg/log/log.go (hunk 2 above)
Comment (kale-amruta) [reply pkg/log/log.go:132]: @PrasadG193 What if we want to set it explictly in code, using SetLevel() rather than using the env variable?

**Review by pavannd1 — APPROVED** [review]
LGTM 🚀

**Pull request merged**